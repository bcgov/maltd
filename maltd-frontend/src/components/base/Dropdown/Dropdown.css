#dropdown {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  background-color: white;
  color: black;
  padding-right: 1px;
  margin-right: 1px;
  width: 12em;
}

.img {
  width: 20%;
  margin-right: 1px;
}

#title {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin-right: 1px;
  width: 80%;
  padding-right: 1px;
}

@media only screen and (min-width: 880px) and (max-width: 1120px) {
  #dropdown {
    max-width: 8.5em;
  }
}

@media only screen and (min-width: 768px) and (max-width: 879px) {
  #dropdown {
    max-width: 7.5em;
  }
}

@media only screen and (min-width: 451px) and (max-width: 767px) {
  #dropdown {
    width: 18em;
  }
}

@media only screen and (max-width: 450px) {
  #dropdown {
    width: 15em;
  }
}

@media only screen and (max-width: 384px) {
  #dropdown {
    max-width: 12em;
  }
}
