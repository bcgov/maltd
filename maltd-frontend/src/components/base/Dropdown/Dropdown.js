import React, { useState } from "react";
import PropTypes from "prop-types";
import {
  UncontrolledDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem
} from "reactstrap";

export default function Dropdown({ dropdown: { items }, onDropdownClick }) {
  const [selectedTitle, setSelectedTitle] = useState("Select a project");

  const dropdownItems = items.map(item => (
    <DropdownItem
      onClick={() => {
        setSelectedTitle(`${item.name} - ${item.type}`);
        onDropdownClick(item);
      }}
      key={item.id}
    >
      {item.name}-{item.type}
    </DropdownItem>
  ));

  return (
    <UncontrolledDropdown>
      <DropdownToggle
        style={{ backgroundColor: "white", color: "black" }}
        caret
      >
        {selectedTitle}
      </DropdownToggle>
      <DropdownMenu>{dropdownItems}</DropdownMenu>
    </UncontrolledDropdown>
  );
}

Dropdown.propTypes = {
  dropdown: PropTypes.shape({
    items: PropTypes.array.isRequired
  }).isRequired,
  onDropdownClick: PropTypes.func.isRequired
};
