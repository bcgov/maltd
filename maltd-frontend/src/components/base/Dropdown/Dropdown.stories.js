import React from "react";
import { storiesOf } from "@storybook/react";
import "bootstrap/dist/css/bootstrap.css";
import Dropdown from "./Dropdown";

const dropdown = {
  defaultTitle: "Choose a project",
  items: ["System1", "System2"]
};

storiesOf("Dropdown", module).add("default", () => (
  <Dropdown dropdown={dropdown} />
));
