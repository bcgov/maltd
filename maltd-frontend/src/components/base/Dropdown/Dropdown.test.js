import React from "react";
import renderer from "react-test-renderer";
import Dropdown from "./Dropdown";

describe("Dropdown", () => {
  test("Component renders as expected", () => {
    const component = renderer.create(
      <Dropdown
        dropdown={{
          items: [
            {
              id: "123",
              name: "System 1",
              type: "Dynamics"
            }
          ]
        }}
        onDropdownClick={() => jest.fn()}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
