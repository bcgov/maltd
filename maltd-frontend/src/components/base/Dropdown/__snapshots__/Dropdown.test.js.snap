// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component renders as expected 1`] = `
<div
  className="dropleft dropdown"
  onKeyDown={[Function]}
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-label={null}
    className="dropdown-toggle btn btn-secondary"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "color": "black",
        "maxWidth": "140px",
        "overflow": "hidden",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }
    type="button"
  >
    Select a project
  </button>
  <div
    aria-hidden={true}
    className="dropdown-menu"
    role="menu"
    tabIndex="-1"
  >
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      System 1
      -
      Dynamics
    </button>
  </div>
</div>
`;
